apps/api/src/app/subscribers/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts

```diff
@@ -18,15 +19,17 @@ export class GetSubscriberTemplatePreference {
   constructor(
     private subscriberPreferenceRepository: SubscriberPreferenceRepository,
     private notificationTemplateRepository: NotificationTemplateRepository,
-    private messageTemplateRepository: MessageTemplateRepository
+    private messageTemplateRepository: MessageTemplateRepository,
+    private subscriberRepository: SubscriberRepository
   ) {}
 
   async execute(command: GetSubscriberTemplatePreferenceCommand): Promise<ISubscriberPreferenceResponse> {
     const activeChannels = await this.queryActiveChannels(command);
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
 
     const subscriberPreference = await this.subscriberPreferenceRepository.findOne({
       _environmentId: command.environmentId,
-      _subscriberId: command.subscriberId,
+      _subscriberId: subscriber !== null ? subscriber._id : command.subscriberId,
```

Comment:
@davidsoderberg why is this possible here? Can;t we make sure that it will always be the subscriberId? This API is a bit vague that one time it's a mongo id and the other not. Wdyt?